File: backend/src/main/java/si/feri/fitlink/coaching/Coaching.java
package si.feri.fitlink.coaching;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Document(collection = "coachings")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Coaching {
    @Id
    private String id;
    private String traineeId;
    private String trainerId;
    private CoachingStatus status;
    private String requestMessage;
    private Instant startedAt;
    private Instant endedAt;
    private String endedBy;
    @Builder.Default
    private List<si.feri.fitlink.checkin.CheckIn> checkIns = new ArrayList<>();
}

File: mobile/src/types/coaching.ts
import { CheckIn } from "./checkin";

export interface CoachingRequest {
  trainerId: string;
  requestMessage: string;
}

export interface Coaching {
  id: string;
  traineeId: string;
  trainerId: string;
  status: 'PENDING' | 'ACTIVE' | 'ENDED' | 'REJECTED' | string;
  requestMessage: string | null;
  startedAt: string | null;
  endedAt: string | null;
  endedBy: string | null;
  checkIns: CheckIn[];
}